### ISSUE TITLE
Usage example?

### ISSUE THREAD

**Issue by atotic** [wnr/element-resize-detector#44]
Hi, I work on Chrome, and my next task is to create a native resize observer. It'll be similar to element-resize-detector.
I am currently writing a spec, and am looking for usage examples. Do you have any examples of where resize detector was needed?

PS: The scrolling trick you've used is neat. I've thought about how I'd write a polyfill, and did not come up with anything as clever.

**Comment by wnr** [comment]
Hi,

That's great news!

Yes, me and my team are currently developing a new responsive version of the web-based bank application for one of the top banks in Sweden. The app currently consists of 3000 pages, and there are several sister-apps to fill different purposes. We wanted to create reusable view modules that can be shared between pages/apps, and therefore we are much interested in element queries. While waiting for element queries to be standardised, we created a JS library to help us with this task called ELQ (https://github.com/elqteam/elq) that of course uses the element-resize-detector in order to detect resize events of elements.

In practice, it means that we do not use CSS media queries at all in order to stay modular. So all our views (tables, forms, menus, etc.) all use element queries and indirect element resize detection in order to know when to change layout. I can provide you tons of specific use cases for our views, If you want.

While modular views using element queries have been the main use case, we have also used the element resize detector in documentation purposes. All our modular views can be viewed in a special documentation app that we created for our customers. In the app, it is possible to see the view live with different data and settings. Also, we visualise the responsivity by showing a ruler marked with breakpoints of the view. The ruler uses element resize detection in order to adapt itself to the container. Also, the user is able to resize views freely to see the responsivity in action. Here, we also use the element resize detection to display the current size of the view for the user.

This is an overview of our usage of element resize detection. I could provide you with more details if you like. We've created an article that might be interesting to you regarding our usage of element queries [https://github.com/wnr/chi-elq-note/raw/springer/elq.pdf](https://github.com/wnr/chi-elq-note/raw/springer/elq.pdf).

> PS: The scrolling trick you've used is neat. I've thought about how I'd write a polyfill, and did not come up with anything as clever.

Thank you! I cannot take full credit as Daniel Buchner originally came up with the idea. However, I do have improved it quite a bit :)

PS: I would love to assist with the spec/polyfill :)
